### ISSUE TITLE
[BUG] Cannot pass null for nullable fields

### ISSUE THREAD

**Issue by doublemc** [iway1/trpc-panel#39]
I cannot find a way to pass null for nullable field. When I type in `null` then it's sent as string "null".
Additionally when string field is cleared then it's sent as empty string `""` instead of undefined / null. Booleans / arrays behave the same - they have default false/empty array and cannot pass null / undefined.

It would be good if there was an option to choose if I want to send it as null or undefined - maybe some checkbox?

Those issues are easily reproducible so I don't have any code example but if you need it then let me know and I will prepare something.

**Comment by rafalsobota** [comment]
In my view, it would be fine if I could write just `null` for null and `"null"` for the string "null" which is a rare case.

**Comment by iway1** [comment]
Yeah this would be a useful feature to add for sure, definitely isn't uncommon to need to pass null values for testing and such. I can't really think of any approaches other than a checkbox so I think that might be the way to do it.

As far as setting a field to "undefined" after entry, that's actually impossible in `react-hook-form`, which this library depends on for form state management. So we can't really do that without some hackiness, I think it's better just to leave as is.

We can set default values for nullable fields to null, though, which would make them reset to null when cleared since react hook form allows that.. That could be included as part of the change for the null checkbox implementation 😄

**Comment by dvv** [comment]
Hi!

I can't deal properly with the following "patch" schema commonly used in my setup:
```zod
z.object({ arrayField: z.array(z.string()).optional().nullable() }) // states: valid array | null | undefined
```
This one is sent as `{ "arrayField": [] }` if the panel form is left untouched.
Can it be changed to not include `.optional().nullable()` fields if form is untouched?

If this is breaking where to look at the code to tune the behavior?

TIA
